src/programs.c

```diff
@@ -2117,60 +2479,6 @@ int kretprobe__ret_sys_clone3(struct pt_regs *ctx)
     return exit_clone3(ctx);
 }
 
-// This probe can generically read the pid from a task_struct at any point
-// where the first argument is a pointer to a task_struct, the event emit
-// is a RETCODE with the correct PID, intended for use with tracing fork,
-// clone, etc.
-SEC("kprobe/read_pid_task_struct")
-int kprobe__read_pid_task_struct(struct pt_regs *ctx)
-{
-    // get new current
-    void *ts = (void *)PT_REGS_PARM1(ctx);
-
-    // get the true pid
-    u32 pid = 0;
-    u32 tgid = 0;
-    read_value(ts, CRC_TASK_STRUCT_PID, &pid, sizeof(pid));
-    read_value(ts, CRC_TASK_STRUCT_TGID, &tgid, sizeof(tgid));
-    u64 pid_tgid = (u64) tgid << 32 | pid;
-
-    // get the real parent
-    read_value(ts, CRC_TASK_STRUCT_REAL_PARENT, &ts, sizeof(ts));
-    u32 ppid = 0;
-    u32 ptgid = 0;
-
-    // find ppid and ptid (offsets)
-    // ts->real_parent->pid
-    read_value(ts, CRC_TASK_STRUCT_PID, &ppid, sizeof(ppid));
-    // ts->real_parent->tgid
-    read_value(ts, CRC_TASK_STRUCT_TGID, &ptgid, sizeof(ptgid));
-
-    // combine to find ID, get ID
-    u64 ppid_tgid = (u64) ptgid << 32 | ppid;
-    u64 *id = bpf_map_lookup_elem(&telemetry_ids, &ppid_tgid);
-
-    if (!id)
-        return -1;
-
-    // send event with ID
-    ptelemetry_event_t ev = &(telemetry_event_t){
-        .id = 0,
-        .done = TRUE,
-        .telemetry_type = 0,
-        .u.v = {
-                .value[0] = '\0',
-                .truncated = FALSE,
-                },
-        };
-
-    ev->id = *id;
-    bpf_map_delete_elem(&telemetry_ids, &ppid_tgid);
-    ev->telemetry_type = TE_RETCODE;
-    ev->u.r.pid_tgid = pid_tgid;
-    push_telemetry_event(ctx, ev);
-    return 0;
```

Comment:
Please put this back, I'm using it. 😆

Reply:
Oh that's weird. I don't remember touching that code at all. Not sure how that got deleted

Reply:
Git probably happily nuked the change in a merge or rebase.